### PR TITLE
Warn instead of fail on failure to validate against DB

### PR DESCRIPTION
**Pull request by mheon**
When the database has somehow stored the empty string for a field we wish to validate against, we cannot perform any reasonable validation. Instead of making this a hard failure, print a warning that we cannot validate, and bad things may happen.

### CODE REVIEW

File: libpod/boltdb_state_internal.go [hunk 1]
```diff
@@ -146,10 +146,11 @@ func validateDBAgainstConfig(bucket *bolt.Bucket, fieldName, runtimeValue string
...
 			}
 
-			// If DB value is the empty string, check that the
-			// runtime value is the default
-			if string(keyBytes) == "" && defaultValue != "" &&
-				runtimeValue == defaultValue {
+			// If DB value is the empty string, we ought to be fine.
+			// (Though I'm honestly not sure how we got this far
+			// with an empty-string DB config)
+			if string(keyBytes) == "" {
```
Comment (rhatdan) [inline libpod/boltdb_state_internal.go]: Should we update the data with the correct value?

**Comment by rhatdan** [comment]
One question, but I am fine with this change.

File: libpod/boltdb_state_internal.go (hunk 1 above)
Comment (mheon) [reply libpod/boltdb_state_internal.go]: The problem is that we don't really know the correct value here - the DB is supposed to know, and it doesn't, so we can only hope that we're using the right storage driver.

**Comment by giuseppe** [comment]
LGTM

File: libpod/boltdb_state_internal.go (hunk 1 above)
Comment (mheon) [reply libpod/boltdb_state_internal.go]: I think I've figured out why we're entering "" in the first place though. Going to fix that next.

**Comment by mheon** [comment]
Added one more commit to not unconditionally overwrite storage defaults on a configuration being present when we run as root - I think this might resolve our empty-string storage driver issues

**Comment by TomSweeneyRedHat** [comment]
LGTM

File: pkg/util/utils.go [hunk 2]
```diff
@@ -297,6 +297,9 @@ func GetDefaultStoreOptions() (storage.StoreOptions, error) {
 		err                      error
 	)
 	storageOpts := storage.DefaultStoreOptions
+	if !rootless.IsRootless() {
```
Comment (rhatdan) [inline pkg/util/utils.go]: This code makes little sense.  Above you set storageOpts to DefaultStoreOptions, then you overwrite it in both root and rootless mode.

**Comment by rhatdan** [comment]
@mheon Any update on this PR?

**Comment by mheon** [comment]
@rhatdan I think we're still good to proceed, unless this code already moved out into c/storage

**Comment by TomSweeneyRedHat** [comment]
@mheon, you still have a lingering comment here from @rhatdan

**Comment by TomSweeneyRedHat** [comment]
and you need to merge too apparently

**Comment by mheon** [comment]
I'm having second thoughts about this one, might just close it...

**Comment by mheon** [comment]
This isn't worth doing, I think

**Pull request closed without merging**